src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/Diagnostics/DiagnosticHandler.cs

```diff
@@ -0,0 +1,310 @@
+                result.Add(DiagnosticTag.Unnecessary);
+
+            if (diagnosticData.CustomTags.Contains(WellKnownDiagnosticTags.EditAndContinue))
+                result.Add(VSDiagnosticTags.EditAndContinueError);
+
+            return result.ToArray();
+        }
+
+        protected override ValueTask<ImmutableArray<IDiagnosticSource>> GetOrderedDiagnosticSourcesAsync(RequestContext context, CancellationToken cancellationToken)
+        {
+            return ValueTaskFactory.FromResult(GetRequestedDocument(context));
+        }
+
+        protected override VSInternalDiagnosticReport[]? CreateReturn(BufferedProgress<VSInternalDiagnosticReport> progress)
+        {
+            return progress.GetValues();
+        }
+
+        internal static ImmutableArray<IDiagnosticSource> GetRequestedDocument(RequestContext context)
+        {
+            if (context.AdditionalDocument == null)
+            {
+                return ImmutableArray<IDiagnosticSource>.Empty;
+            }
+
+            if (!context.IsTracking(context.AdditionalDocument.GetURI()))
+            {
+                return ImmutableArray<IDiagnosticSource>.Empty;
+            }
+
+            return ImmutableArray.Create<IDiagnosticSource>(new DocumentDiagnosticSource(context.AdditionalDocument));
+        }
+
+        private record struct DocumentDiagnosticSource(TextDocument Document) : IDiagnosticSource
+        {
+            public ProjectOrDocumentId GetId() => new(Document.Id);
+
+            public Project GetProject() => Document.Project;
+
+            public Uri GetUri() => Document.GetURI();
+
+            public async Task<ImmutableArray<DiagnosticData>> GetDiagnosticsAsync(IDiagnosticAnalyzerService diagnosticAnalyzerService, RequestContext context, DiagnosticMode diagnosticMode, CancellationToken cancellationToken)
+            {
+                var document = context.AdditionalDocument;
+                Contract.ThrowIfNull(document);
+
+                var workspace = document.Project.Solution.Workspace;
+
+                var filePath = document.FilePath;
+                Contract.ThrowIfNull(filePath);
+
+                var optionSet = workspace.Options;
+                var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                var diagnostics = new List<DiagnosticData>();
+
+                var definedSettings = new HashSet<string>();
+
+                foreach (var line in text.Lines)
+                {
+                    var lineText = line.ToString();
+
+                    // Check that setting definition doesn't have more than 1 '='
+                    if (lineText.Where(c => c == '=').Count() > 1)
+                    {
+                        diagnostics.Add(CreateDiagnosticData(document, line, EditorConfigDiagnosticIds.IncorrectSettingDefinition, DiagnosticSeverity.Error, false, 1, Document.Project.Id));
+                    }
```

Comment:
This check looks fragile.

- `insert_final_newline = true # comment contains =`
- I'm not sure if it's possible that a string value can contain `=` or not (and does it require escape? I don't know).
- I'm not sure if it's possible to specify `[a==b.cs]` (it's weird to have a file with `=` in its name, but possible?) (does it require escape? I don't know)